lib/RestService/src/RestService.h

```diff
@@ -0,0 +1,312 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  REST service
+ * @author Niklas Kümmel
+ *
+ * @addtogroup REST_SERVICE
+ *
+ * @{
+ */
+
+#ifndef REST_SERVICE_H
+#define REST_SERVICE_H
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include <IService.hpp>
+#include <AsyncHttpClient.h>
+#include <TaskProxy.hpp>
+#include <Mutex.hpp>
+#include <ArduinoJson.h>
+#include <map>
+#include <HttpStatus.h>
+#include <Logging.h>
+#include <vector>
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+
+/**
+ * The REST service handles outgoing REST-API calls and their responses.
+ */
+class RestService : public IService
+{
+public:
+
+    /**
+     * callback prototype.
+     */
+
+    /**
+     * Get the REST service instance.
+     *
+     * @return REST service instance
+     */
+    static RestService& getInstance()
+    {
+        static RestService instance; /* idiom */
+
+        return instance;
+    }
+
+    /**
+     * Start the service.
+     *
+     * @return If successful started, it will return true otherwise false.
+     */
+    bool start() final;
+
+    /**
+     * Stop the service.
+     */
+    void stop() final;
+
+    /**
+     * Process the service.
+     */
+    void process() final;
+
+    /**
+     * Get restId for plugin
+     *
+     * @return Unique restId
+     */
+    int registerPlugin();
+
+    /**
+     * Set filter, only one per plugin
```

Comment:
Complete sentences please ... dot is missing at the end. The same for other places as well.